src/entities/Proposal/model.ts

```diff
@@ -247,7 +252,10 @@ export default class ProposalModel extends Model<ProposalAttributes> {
         )}
     WHERE "deleted" = FALSE 
     ${conditional(!!filter.user && !filter.coauthor, SQL`AND p."user" = ${filter.user}`)} 
-    ${conditional(!!filter.coauthor, SQL`AND lower(c."address") = lower(${filter.user})`)}
+    ${conditional(
+      !!filter.coauthor,
+      SQL`AND lower(c."address") = lower(${filter.user}) AND (CASE WHEN p."finish_at" < NOW() THEN c."status" IN (${CoauthorStatus.APPROVED}, ${CoauthorStatus.REJECTED}) ELSE TRUE END)`
```

Comment:
🧠

Reply:
I don't think this part can be encapsulated since it depends on the finish date of each proposal 🤔